### ISSUE TITLE
Cant disable HUD

### ISSUE THREAD

**Issue by MisterE123** [minetest/minetest#11075]
##### Summary
I want to make a blindness statuseffect. So, I add an HUD that covers the screen. That works. But then, someone pointed out that you can disable HUD. Then someone suggested to make a 1st person entity that covered the view, and then pointed out 3rd person circumvents this. I would like servers to have more control over players' controls. Its not like this is a security issue: in fact its just the opposite: for features that require limiting player abilities, not giving that control to servers allows cheating easily, and makes this feature impossible to implement

**Comment by Wuzzy2** [comment]
You can disable most of the relevant HUD already, look for `hud_flags` in the documentation. You can disable (server-side):

* Minimap
* Hotbar
* Crosshair
* Wieldhand

You can NOT disable:
* Chat
* That info text that appears when you enable e.g. fly mode
* Debug text (not a problem IMO)

Is that good enough for you or do you need more features?

**Comment by hlqkj** [comment]
I think they need a way to force an hud element to be visible/can't be hidden by the player with clients controls... Not sure it can be done now?

**Comment by MisterE123** [comment]
sorry, I added the wrong title: I was tired. The problem is that the client *Can* disable the HUD, but I need to force the client to have the HUD. I want to make the client blind (by covering their screen). So I have done that with an HUD, bu they can easily press F1 and they have no problem. I wanted a status effect of blindness, given by a mob.

**Comment by Wuzzy2** [comment]
Ah, now it makes sense. Good suggestion. :+1:

So one solution to this problem would be to add a new parameter for HUD elements to force-show it at all times, even if the player pressed F1 to hide the HUD. This will be useful for “fake HUD images” like the big black image that you suggested.

The `vignette` mod would also benefit from this, because right now, the vignette also disappears if you press F1.

**Comment by appgurueu** [comment]
Another hack I've seen is a particlespawner attached to the player and only visible to the player spawning multiple huge particles.

**Comment by Zughy** [comment]
UP

Labelling this issue as "high priority" because without this games can't have a saying on when the player cannot see if they know the F1 trick. This is true for an intro like the one in Glitch but also for the blindness effect in the Murder minigame: player can avoid the malus and basically cheat. Maybe after a certain `z_index` HUDs shouldn't disappear?

Yes, there might be hacks but frankly it should be available without such hacks

**Comment by Wuzzy2** [comment]
Yeah, I'm fine with making all HUD elements with a `z_index` of 2000 or above non-removable. I don't think a game used such a high `z_index` so far.

**Comment by Zughy** [comment]
The issue with that is, vignette HUDs won't fit into the value, because MT docs suggest to put them at -500

**Comment by sfan5** [comment]
Just add an extra flag for "unhideable" or whatever?

**Comment by Zughy** [comment]
@sfan5 makes sense

**Comment by Wuzzy2** [comment]
Proposal copied from #17072:

> Add a new field to HUD definitions (or re-use an existing field, if possible) called `can_hide`. This is a boolean. If true, this HUD element will be hidden with the Toggle HUD key, if false, it will always be shown and the Toggle HUD key does nothing. The default is false.
